src/Shell/TrashShell.php

```diff
@@ -0,0 +1,169 @@
+ *
+ */
+class TrashShell extends Shell
+{
+    /**
+     * Get the option parser for this shell.
+     *
+     * @return \Cake\Console\ConsoleOptionParser
+     */
+    public function getOptionParser()
+    {
+        $parser = parent::getOptionParser();
+        $option = [
+                    'age' => [
+                        'help' => __('Time to delete (ex: -1 day / -1 months)'),
+                        'required' => false,
+                        'short' => 'a'
+                    ],
+                    'dry-run' => [
+                        'help' => __('Permanently delete soft-deleted record)'),
+                        'required' => false,
+                        'boolean' => true
+                    ]
+                ];
+        $parser->addSubcommand('list_modules', [
+            'help' => 'Show a list of all available CSV modules.',
+            'required' => false
+        ]);
+        $parser->addSubcommand('clear_all', [
+            'help' => 'Clear all trashed records.',
+            'parser' => [
+                'description' => [
+                    'Clear the trashed records for the supperted modules.',
+                    'Use `cake cache list_modules` to list available CSV modules'
+                ],
+                'options' => $option
+            ]
+        ]);
+        $parser->addSubcommand('clear', [
+            'help' => 'Clear the trashed records for a specified module.',
+            'parser' => [
+                'description' => [
+                    'Clear the trashed records for a specified module.',
+                    'For example, `cake trash clear _cake_module_` will clear the module trashed recors',
+                    'Use `cake cache list_modules` to list available CSV modules'
+                ],
+                'arguments' => [
+                    'prefix' => [
+                        'help' => 'The cache prefix to be cleared.',
+                        'required' => false
+                    ]
+                ],
+                'options' => $option
+            ]
+        ]);
+
+        return $parser;
+    }
+
+    /**
+     * List of all CSV Modules
+     * @return void
+     */
+    public function listModules()
+    {
+        $modules = $this->getList();
+        foreach ($modules as $value) {
+            $this->out($value);
+        }
+    }
+
+    /**
+     * If the argument 'force' is not set, it will display the trash info about the selected modules
+     * @param string $module Module to analyse/clean
+     * @return void
+     */
+    public function clear($module = null)
+    {
+        if (!$module) {
+            $this->out('No module selected');
+
+            return;
+        }
+
+        $query = TableRegistry::get($module);
+
+        if (!$query->behaviors()->has('Trash')) {
+            // Module don't support trash
+            return;
+        }
+
+        try {
+            $trashEntities = TableRegistry::get($module)->find('onlyTrashed')->toArray();
```

Comment:
@eliamariutti instead of retrieving the whole dataset we can only count the records. Suggestion follows:
```suggestion
            $query = TableRegistry::get($module)->find('onlyTrashed');
            $total = $query->select(['total' => $query->func()->count('*')])->first()->toArray();
```

Also the above should be executed only and only if we are running in a `dry-run` mode (since we are not using `$trashEntities` anywhere else.